src/parser/warrior/fury/modules/talents/SuddenDeath.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import { formatPercentage, formatThousands } from 'common/format';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import Analyzer from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
+
+/**
+ * Your attacks have a chance to reset the cooldown of Execute
+ * and make it usable on any target, regardless of their health.
+ */
+
+class SuddenDeath extends Analyzer {
+  suddenDeathProcs = 0;
+  suddenDeathProcsUsed = 0;
+  suddenDeathDamage = 0;
+
+  lastExecuteCast = 0;
+  lastSuddenDeathExecuteCast = 0;
+
+  constructor(...args) {
+    super(...args);
+
+    this.active = this.selectedCombatant.hasTalent(SPELLS.SUDDEN_DEATH_TALENT_FURY.id);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EXECUTE_FURY), this.onExecuteCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell([SPELLS.EXECUTE_DAMAGE_FURY, SPELLS.EXECUTE_DAMAGE_OH_FURY]), this.onExecuteDamage);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.SUDDEN_DEATH_TALENT_FURY_BUFF), this.onSuddenDeathProc);
+    this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.SUDDEN_DEATH_TALENT_FURY_BUFF), this.onSuddenDeathProc);
+  }
+
+  onExecuteCast(event) {
+    if (this.selectedCombatant.hasBuff(SPELLS.SUDDEN_DEATH_TALENT_FURY_BUFF.id)) {
+      this.lastSuddenDeathExecuteCast = event.timestamp;
+      this.suddenDeathProcsUsed += 1;
+    }
+    this.lastExecuteCast = event.timestamp;
+  }
+
+  onExecuteDamage(event) {
+    if (this.lastExecuteCast === this.lastSuddenDeathExecuteCast) {
+      this.suddenDeathDamage += event.amount + (event.absorbed || 0);
+    }
+  }
+
+  onSuddenDeathProc() {
+    this.suddenDeathProcs += 1;
+  }
+
+  get damagePercent() {
+    return this.owner.getPercentageOfTotalDamageDone(this.suddenDeathDamage);
```

Comment:
Just in general, this metric is not very accurate. Calculating the DPS gain from this trait is quite complicated, as you need to not only count the extra procs on targets above the normal health threshold, you also need to take into account any cooldown reduction it gives you on those below the threshold. Instead of the complicated DPS, maybe two different metrics:
`Extra 'high HP' executes (and you could give the damage values for those) on targets _above_ the standard health threshold`
+
`Cooldown reduction (for targets that were below the health threshold anyway but when you reduced the CD; again requires looking at SpellUsable)`

(In reality, this metric currently overvalues the trait, because it counts all executes, even those on targets below 20% hp, as fully related to the trait even if all the trait did was reduce the CD by a second)